File: sdks/ts/src/sdk/models/operations/postzipdownloads.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PostZipDownloadsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: shared.ZipDownloadRequest;
}


export class PostZipDownloadsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  zipDownload?: shared.ZipDownload;
}

File: sdks/python/src/boxsdk/models/operations/post_zip_downloads.py
import dataclasses
from typing import Optional
from boxsdk import utils
from ..shared import zipdownloadrequest as shared_zipdownloadrequest
from ..shared import clienterror as shared_clienterror
from ..shared import zipdownload as shared_zipdownload


@dataclasses.dataclass
class PostZipDownloadsRequest:
    request: Optional[shared_zipdownloadrequest.ZipDownloadRequest] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    retries: Optional[utils.RetryConfig] = dataclasses.field(default=None)
    

@dataclasses.dataclass
class PostZipDownloadsResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    client_error: Optional[shared_clienterror.ClientError] = dataclasses.field(default=None)
    zip_download: Optional[shared_zipdownload.ZipDownload] = dataclasses.field(default=None)
    
